### PR TITLE
EVG-7972 make command optional for containers spawned from task

### PR DESCRIPTION
**Pull request by ablack12**
I'll update the docs that the container will run the image's default entrypoint if a command is not specified.

### CODE REVIEW

**Comment by ablack12** [comment]
I removed the user checks for the routes relevant to docker;  @julianedwards added them in https://github.com/evergreen-ci/evergreen/commit/fad8567deddd2289138f5b5ab12aac1a4bf20d47 but we can't enforce a user for these commands to work.

File: rest/route/service.go [hunk 1]
```diff
@@ -102,9 +102,9 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/hosts/{host_id}/change_password").Version(2).Post().Wrap(checkUser).RouteHandler(makeHostChangePassword(sc, env))
 	app.AddRoute("/hosts/{host_id}/extend_expiration").Version(2).Post().Wrap(checkUser).RouteHandler(makeExtendHostExpiration(sc))
 	app.AddRoute("/hosts/{host_id}/terminate").Version(2).Post().Wrap(checkUser).RouteHandler(makeTerminateHostRoute(sc))
-	app.AddRoute("/hosts/{host_id}/status").Version(2).Get().Wrap(checkUser).RouteHandler(makeContainerStatusManager(sc))
-	app.AddRoute("/hosts/{host_id}/logs/output").Version(2).Get().Wrap(checkUser).RouteHandler(makeContainerLogsRouteManager(sc, false))
-	app.AddRoute("/hosts/{host_id}/logs/error").Version(2).Get().Wrap(checkUser).RouteHandler(makeContainerLogsRouteManager(sc, true))
+	app.AddRoute("/hosts/{host_id}/status").Version(2).Get().RouteHandler(makeContainerStatusManager(sc))
```
Comment (john-m-liu) [inline rest/route/service.go:105]: We need some sort of authentication on these routes. You can maybe reuse the host check that some of the agent routes use

Comment (ablack12) [reply rest/route/service.go]: Adding something, although I won't be able to test until I get staging access.

**Comment by ablack12** [comment]
Created new PR https://github.com/evergreen-ci/evergreen/pull/3581 with cherry-picked commits, closing.

**Pull request closed without merging**